src/content/datachannel/messaging/main.js

```diff
@@ -0,0 +1,168 @@
+/*
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+import {LitElement, html} from 'https://unpkg.com/@polymer/lit-element?module';
+
+class MessagingSample extends LitElement {
+  constructor() {
+    super();
+    this._localConnection = this._remoteConnection = this._localChannel = this._remoteChannel = null;
+  }
+
+  disconnect(e) {
+    this._root.querySelector('#disconnectButton').disabled = true;
+    this._localConnection.close();
+    this._remoteConnection.close();
+  }
+
+  async connect(e) {
+    try {
+      this._root.querySelector('#connectButton').disabled = true;
+      const dataChannelParams = {ordered: false};
+      let lpc = this._localConnection = new RTCPeerConnection(null);
+      lpc.addEventListener('icecandidate', (e) => this._onIceCandidate(e, this._remoteConnection));
+      let rpc = this._remoteConnection = new RTCPeerConnection(null);
+      rpc.addEventListener('icecandidate', (e) => this._onIceCandidate(e, this._localConnection));
+
+      let lc = this._localChannel = this._localConnection
+        .createDataChannel('messaging-channel', dataChannelParams);
+      lc.binaryType = 'arraybuffer';
+      lc.addEventListener('open', (e) => this._localChannelOpen(e));
+      lc.addEventListener('close', (e) => this._localChannelClosed(e));
+      lc.addEventListener('message', (e) => this._onLocalMessageReceived(e));
+
+      rpc.addEventListener('datachannel', (e) => this._onRemoteDataChannel(e));
+
+      const localOffer = await lpc.createOffer();
+      console.log(`Got local offer ${JSON.stringify(localOffer)}`);
+      lpc.setLocalDescription(localOffer);
+      rpc.setRemoteDescription(localOffer);
+      const remoteOffer = await rpc.createAnswer();
+      console.log(`Got remote answer ${JSON.stringify(remoteOffer)}`);
+      rpc.setLocalDescription(remoteOffer);
+      lpc.setRemoteDescription(remoteOffer);
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  async _onIceCandidate(e, connection) {
+    try {
+      console.log(`onIceCandidate: ${JSON.stringify(e)}`);
+      // eslint-disable-next-line no-unused-vars
+      const result = await connection.addIceCandidate(e.candidate);
+      console.log('addIceCandidate successful!');
+    } catch (e) {
+      console.log(`error on addIceCandidate: ${JSON.stringify(e)}`);
+    }
+  }
+
+  _localChannelOpen(event) {
+    console.log(`Local channel open: ${JSON.stringify(event)}`);
+    this._root.querySelector('#sendLocal').disabled = false;
+    this._root.querySelector('#sendRemote').disabled = false;
```

Comment:
this doesn't listen for the remote channel open event. Now its pretty hard to get this to fail in this sample but still wrong semantics

Reply:
True. I'll add the same to the remote as well for clarity.